docs(env-and-mode): add override rule for the env (#6304)
vitejs/vite: docs/guide/env-and-mode.md

## `.env` Files

> Vite uses [dotenv](https://github.com/motdotla/dotenv) to load additional environment variables from the following files in your [environment directory](/config/#envdir) (if define the same variables will override variables in the following reverse order):

Reviewer: IMO it reads hard :thinking: 
Maybe better something like this? Could even be written better.

```suggestion
Vite uses [dotenv](https://github.com/motdotla/dotenv) to load additional environment variables from the following files in your [environment directory](/config/#envdir) (if they define the same variable, the variable defined in the most specific file will win):
```

Author: I also think `.env.${mode}.local` is in the most specific, but it will be overwrite 😅

Author: in not the most specific will win?

Reviewer: If the most specific currently don't win, then IMO this is a bug and should be changed
And I assume this is what #6306 is fixing :thinking:

Reviewer 2: check https://cli.vuejs.org/guide/mode-and-env.html#environment-variables, see green alert with title `Env Loading Priorities`

.env.[mode].local   # only loaded in specified mode, ignored by git
```

:::tip Env Loading Priorities

An env file for a specific mode (e.g. `.env.production`) will take higher priority than a generic one (e.g. `.env`).

> In addition, environment variables that already exist when vite is executed have the highest priority and will not be overwritten by .env files.

Reviewer 3: ```suggestion
In addition, environment variables that already exist when Vite is executed have the highest priority and will not be overwritten by `.env` files.
```

> .env files are loaded at the start of vite. Restart the service after making changes.

Reviewer 3: ```suggestion
`.env` files are loaded at the start of Vite. Restart the server after making changes.
```
